src/storage/src/hummock/sstable/xor_filter.rs

```diff
@@ -158,29 +281,32 @@ impl XorFilterReader {
             };
         }
 
-        let kind = *buf.last().unwrap();
-        let filter = if kind == FOOTER_XOR16 {
-            let buf = &mut &buf[..];
-            let xor_filter_seed = buf.get_u64_le();
-            let xor_filter_block_length = buf.get_u32_le();
-            // is correct even when there is an extra 0xff byte in the end of buf
-            let len = buf.len() / 2;
-            let xor_filter_fingerprints = (0..len)
-                .map(|_| buf.get_u16_le())
-                .collect_vec()
-                .into_boxed_slice();
-            XorFilter::Xor16(Xor16 {
-                seed: xor_filter_seed,
-                block_length: xor_filter_block_length as usize,
-                fingerprints: xor_filter_fingerprints,
-            })
+        let kind = *data.last().unwrap();
+        let l = data.len();
+        let filter = if kind == FOOTER_BLOCKED_XOR16 {
```

Comment:
Encapsulate the new function 

- to block_xor_16
- to xor_16
- to xor_8

Reply:
done